### ISSUE TITLE
have you train this on the stereodataset?

### ISSUE THREAD

**Issue by meteorshowers** [huyaoyu/NewStereo#1]
have you trained this on the stereo dataset?
how about the performance?
thanks!

**Comment by huyaoyu** [comment]
The performance is not as good as the PSMNet or my modified version of PSMNet, in terms of speed and absolute accuracy. However, as you may have already noticed, I am working on making things memory efficient. By using the correlation layer from the PWCNet, the memory footprint is lowered for the disparity estimation without the final disparity refinement. But the refinement layer may cause memory trouble if it has lots of layers and has to be working on a higher-level resolution. I am still working on this.

**Comment by meteorshowers** [comment]
I am a little confused
pwcnet claims the speed is 30fps, psmnet's speed is 2fps.
In terms of speed, pwcnet is more efficient, I think?

**Comment by meteorshowers** [comment]
@huyaoyu

**Comment by huyaoyu** [comment]
> I am a little confused
> pwcnet claims the speed is 30fps, psmnet's speed is 2fps.
> In terms of speed, pwcnet is more efficient, I think?

Yes, you are right. PWCNet's speed is good. What I meant is that my current model, which is using components from the PWCNet, is not that fast. Cause I am adding other stuff. 

Trained on the SceneFlow dataset. If we compare accuracy, PSMNet achieves better performance than what I currently get on the evaluation set with the same amount of training without any data augmentation.

**Comment by meteorshowers** [comment]
Thanks!
Have you evaluated the pwcnet's performance on the stereo matching task?

**Comment by huyaoyu** [comment]
> Thanks!
> Have you evaluated the pwcnet's performance on the stereo matching task?

I only have my own implementation of the PWCNet which could get less than 2.0 EPE error on the roughly 1/4-sized image (1/2 height x 1/2 width) of the scene flow dataset after 2-3 training epochs. PWCNet is originally designed for optical flow, once used to estimate stereo disparity, the performance is not as good as other SOTA stereo models. I have to use a larger correlation range and a larger kernel size. However, the accuracy is not my primary concern, the GPU memory is. For a model such as the PSMNet, the memory consumption is much larger than the structures adopted by models like the PWCNet.
